backend.go

```diff
@@ -301,19 +311,24 @@ func (b *versionedKVBackend) getKeyEncryptor(ctx context.Context, s logical.Stor
 
 // config takes a storage object and returns a configuration object
 func (b *versionedKVBackend) config(ctx context.Context, s logical.Storage) (*Configuration, error) {
+	cached := b.globalConfig.Load().(*Configuration)
```

Comment:
Would it be better to use a mutex here so that only one request will go to disk for the config?